Modal: add `size` prop to support a selection of preset modal sizes (#54471)
WordPress/gutenberg: packages/components/src/modal/README.md

-   Required: No
-   Default: `false`

#### `contentWidth`: `'small' | 'medium' | 'large'`

If this property is added, it will constrain the `max-width` of the modal's contents, preventing it from growing too wide. This prop only applies when `isFullScreen` is `false`.

> 

Reviewer: Missing `Required: No`

Author: Yeah I was... testing you. See if you noticed. Yeah. You nailed it, congrats. 😆

-   Required: No
-   Default: `false`

#### `contentWidth`: `'small' | 'medium' | 'large'`

If this property is added, it will constrain the `max-width` of the modal's contents, preventing it from growing too wide. This prop only applies when `isFullScreen` is `false`.

-   Required: No

> Note: `Modal`'s width can also be controlled by adjusting the width of the modal's contents, or using the `style` prop to set a specific `max-width`.

Reviewer 2: ```suggestion
Note: `Modal`'s width can also be controlled by adjusting the width of the modal's contents via CSS.
```

-   Required: No
-   Default: `false`

> #### `size`: `'small' | 'medium' | 'large'`

Reviewer 2: ```suggestion
#### `size`: `'small' | 'medium' | 'large' | 'fill'`
```

> If this property is added it will cause the modal to render at a preset width, or expand to fill the screen.

Reviewer 2: Let's add a note (both in the README and in the types JSDocs) that the `isFullScreen` prop will take precedence over this prop when set.
